sessionctx/binloginfo/binloginfo.go

```diff
@@ -150,15 +154,45 @@ func SetDDLBinlog(client *pumpcli.PumpsClient, txn kv.Transaction, jobID int64,
 
 const specialPrefix = `/*!90000 `
 
-func addSpecialComment(ddlQuery string) string {
+// AddSpecialComment uses to add comment for table option in DDL query.
+// Export for testing.
+func AddSpecialComment(ddlQuery string) string {
 	if strings.Contains(ddlQuery, specialPrefix) {
 		return ddlQuery
 	}
-	loc := shardPat.FindStringIndex(strings.ToUpper(ddlQuery))
-	if loc == nil {
-		return ddlQuery
+	return addSpecialCommentByRegexps(ddlQuery, shardPat, preSplitPat)
+}
+
+func addSpecialCommentByRegexps(ddlQuery string, regs ...*regexp.Regexp) string {
+	upperQuery := strings.ToUpper(ddlQuery)
+	var specialComments []string
+	minIdx := math.MaxInt64
+	for _, reg := range regs {
```

Comment:
For SQL : `create table t6 (id int ) shard_row_id_bits=2 shard_row_id_bits=2 pre_split_regions=2;`
If it comes with the same `Regexps` in SQL, will the comments duplicate here?  
Should do the check and test.

Reply:
Good catch. Done.